diff --git a/driver/main.cpp b/driver/main.cpp
index 826fc2e9..8a1dadb1 100644
--- a/driver/main.cpp
+++ b/driver/main.cpp
@@ -733,6 +733,8 @@ void registerPredefinedTargetVersions() {
         "S390X"); // For backwards compatibility.
     VersionCondition::addPredefinedGlobalIdent("D_HardFloat");
     break;
+  case llvm::Triple::asmjs:
+    break;
   default:
     error(Loc(), "invalid cpu architecture specified: %s",
           global.params.targetTriple->getArchName().str().c_str());
@@ -835,6 +837,8 @@ void registerPredefinedTargetVersions() {
     VersionCondition::addPredefinedGlobalIdent("AIX");
     VersionCondition::addPredefinedGlobalIdent("Posix");
     break;
+  case llvm::Triple::Emscripten:
+    break;
   case llvm::Triple::UnknownOS:
     if (arch == llvm::Triple::msp430)
       break;
