diff --git a/driver/main.cpp b/driver/main.cpp
index 1472d24a..d0c78fee 100644
--- a/driver/main.cpp
+++ b/driver/main.cpp
@@ -763,6 +763,8 @@ void registerPredefinedTargetVersions() {
         "S390X"); // For backwards compatibility.
     VersionCondition::addPredefinedGlobalIdent("D_HardFloat");
     break;
+  case llvm::Triple::asmjs:
+    break;
   default:
     error(Loc(), "invalid cpu architecture specified: %s",
           triple.getArchName().str().c_str());
@@ -861,6 +863,8 @@ void registerPredefinedTargetVersions() {
     VersionCondition::addPredefinedGlobalIdent("NetBSD");
     VersionCondition::addPredefinedGlobalIdent("Posix");
     break;
+  case llvm::Triple::Emscripten:
+    break;
   case llvm::Triple::OpenBSD:
     VersionCondition::addPredefinedGlobalIdent("OpenBSD");
     VersionCondition::addPredefinedGlobalIdent("Posix");
